components/ArticleItemWidget/ArticleItemWidget.js

```diff
@@ -0,0 +1,242 @@
+    const { id } = this.props;
+    NProgress.start();
+    gql`
+      mutation(
+        $articleId: String!
+        $text: String!
+        $type: ReplyTypeEnum!
+        $reference: String
+      ) {
+        CreateReply(
+          articleId: $articleId
+          text: $text
+          type: $type
+          reference: $reference
+        ) {
+          id
+        }
+      }
+    `({ articleId: id, type: 'NOT_ARTICLE', text }).then(() => {
+      this.handleNotArticleReply(text);
+      this.resetWidgetUI();
+      NProgress.done();
+    });
+  };
+
+  handleReadClick = event => {
+    event && event.preventDefault();
+    this.handleRead();
+    this.resetWidgetUI();
+  };
+
+  handleNotArticleClick = () => {
+    this.setState(({ notArticleSelectionDisplay }) => ({
+      notArticleSelectionDisplay: !notArticleSelectionDisplay,
+    }));
+  };
+
+  componentDidMount = () => {
+    this.preventMobileLongPressMenu();
+  };
+
+  preventMobileLongPressMenu = () => {
+    this.refWidget.addEventListener('contextmenu', event =>
+      event.preventDefault()
+    );
+  };
+
+  preventAll = event => {
+    // stop event cascade (touchstart → touchend → mouseover → mousemove → mousedown → mouseup → click)
+    event.preventDefault();
+    // stop (child touchstart -> parent touchstart,... ) mainly for for parent a:link redirection,
+    event.stopPropagation();
+    return event;
+  };
+
+  // close widget bar and tags extend component
+  resetWidgetUI = () => {
+    this.setState(initialState);
+  };
+
+  openBar = () => {
+    this.setState({
+      barOpen: true,
+      hoverTargetName: 'read', //for mobile, explain in onTouchMoveTip function
+    });
+  };
+
+  hoverTag = event => {
+    const name = event.currentTarget.attributes['name'].value;
+    this.setState({
+      hoverTargetName: name,
+    });
+  };
+
+  unHoverTag = () => {
+    this.setState({
+      hoverTargetName: false,
+    });
+  };
+
+  /**
+   * since the touch start event can only trigger when the finger down at the first tag,
+   * so it's hard to done with like mouse enter/leave strategy
+   * then I use onTouchMove to dynamic check which tag was touch over
+   */
+  onTouchMoveTip = event => {
+    var changedTouch = event.changedTouches[0];
+    var elem = document.elementFromPoint(
+      changedTouch.clientX,
+      changedTouch.clientY
+    );
+    try {
+      var target_name = elem.attributes['name'].value;
+      target_name &&
+        this.setState(
+          Object.assign({}, this.initialTagsState, {
+            [`${target_name}Hover`]: true,
+            hoverTargetName: target_name,
+          })
+        );
+    } catch (error) {
+      // touch move not on the widget tag
+    }
+  };
+
+  // trigger the onClick function in tags config, when finger touch end with tag elm
+  onBarTouchEnd = event => {
+    this.preventAll(event);
+    const target = this.tags.find(select => {
+      return select.name === this.state.hoverTargetName;
+    });
```

Comment:
Since we do not have this.tags for now, we should replace this with a switch statement.

```
switch(this.state.hoverTargetName){
  case 'read': return this.handleReadClick();
  case 'notArticle': return this.handleNotArticleClick();
}
```

Reply:
What a big mistake... thanks~~